### DISCUSSION TITLE
How to create schema from GraphQL File for GraphiQL?

### DISCUSSION THREAD

**Discussion by vedantroy** [graphql/graphiql#1921] [Q&A / Support] [answered]
In production, we disable introspection. But I'd still like to get auto-complete inside of GraphiQL.

As expected, server has access to the `.graphql` file. Is there a way to convert that into a schema, which I can send back to the client, so GraphiQL can have autocomplete?

For context, we're serving GraphiQL as a single HTML file.

Like this:

```js
        ReactDOM.render(
          React.createElement(GraphiQL, { fetcher: fetcher}),
          document.getElementById('graphiql'),
        );

```

**Comment by eins78** [comment, marked as answer]
You can give the schema as a prop (then no introspection will be done).
According to [the docs](https://github.com/graphql/graphiql/blob/main/packages/graphiql/README.md#props) the expected type is [`GraphQLSchema`](https://graphql.org/graphql-js/type/#graphqlschema), so you might need to first convert the `schema.graphql` text file, for example using `buildschema` from the `graphql` js package <https://graphql.org/graphql-js/utilities/#buildschema>.
